File: C++/10000/2001-3000/2749.cpp
// https://www.acmicpc.net/problem/2749 : 피보나치 수 3(c++)
// 2024-05-16
#include <iostream>
#include <string.h>
using namespace std;

int fibonachi(int n)
{
    int result;
    int fibo[1500000];
    fibo[0] = 0;
    fibo[1] = 1;
    for (int i = 2; i < n; i++)
    {
        fibo[i] = fibo[i - 1] + fibo[i - 2];
        fibo[i] %= 1000000;
    }
    result = fibo[n];
    return result;
}

int main(void)
{
    int n;
    cin >> n;
    cout << fibonachi(n % 1500000) << endl;
    return 0;
}

File: Python/10000/2001-3000/2749.py
# https://www.acmicpc.net/problem/2749 : 피본나치 수 3 (python3)
from sys import stdin

input = stdin.readline


def solution():
    mod = 1000000  # 10^6
    fibo = [0, 1]
    p = mod // 10 * 15  # 피사노 주기

    n = int(input())

    for i in range(2, p):
        fibo.append(fibo[i - 1] + fibo[i - 2])
        fibo[i] %= mod

    print(fibo[n % p])


solution()
